lib/api/corsPreflight.js

```diff
@@ -0,0 +1,77 @@
+import { errors } from 'arsenal';
+
+import metadata from '../metadata/wrapper';
+import bucketShield from './apiUtils/bucket/bucketShield';
+import { findCorsRule,
+gatherCorsResHeaders } from './apiUtils/object/corsResponse';
+// import { pushMetric } from '../utapi/utilities';
+
+const requestType = 'objectGet';
+
+const customizedErrs = {
+    corsNotEnabled: 'CORSResponse: CORS is not enabled for this bucket.',
+    notAllowed: 'CORSResponse: This CORS request is not allowed. ' +
+    'This is usually because the evalution of Origin, request method / ' +
+    'Access-Control-Request-Method or Access-Control-Request-Headers ' +
+    'are not whitelisted by the resource\'s CORS spec.',
+};
+
+/** corsPreflight - handle preflight CORS requests
+* @param  {object} request - http request object
+* @param  {function} log - Werelogs request logger
+* @param  {function} callback - callback to respond to http request
+*  with either error code or 200 response
+* @return {undefined}
+*/
+export default function corsPreflight(request, log, callback) {
+    log.debug('processing request', { method: 'corsPreflight' });
+
+    const bucketName = request.bucketName;
+    const corsOrigin = request.headers.origin;
+    const corsMethod = request.headers['access-control-request-method'];
+    const corsHeaders = request.headers['access-control-request-headers'] ?
+        request.headers['access-control-request-headers'].replace(/ /g, '')
+            .split(',').filter(value => value !== '')
```

Comment:
ah, thanks. As discussed it doesn't seem we can do this with .map() but using reduce() to avoid the extra array construction could work. I tried this here: https://github.com/scality/S3/pull/518/commits/18c6f5fcb7d0ad3b518a8e2324ef7ae8c9b539f5